File: backend/src/main/java/com/myfitnesslog/entity/WorkoutStatus.java
package com.myfitnesslog.entity;

/**
 * Lifecycle state of a workout session (DATABASE.md). Persisted by name
 * (EnumType.STRING) to match the workout_session.status CHECK constraint and the
 * Android WorkoutStatus enum.
 */
public enum WorkoutStatus {
    IN_PROGRESS,
    COMPLETED,
    DISCARDED
}

File: android/app/src/main/java/com/myfitnesslog/core/data/local/WorkoutStatus.kt
package com.myfitnesslog.core.data.local

/**
 * Lifecycle state of a workout session (DATABASE.md).
 *
 * A session starts IN_PROGRESS and becomes COMPLETED or DISCARDED exactly once.
 * Completed sessions are immutable history; discarded sessions are retained
 * (not deleted) and simply hidden from the user.
 */
enum class WorkoutStatus {
    IN_PROGRESS,
    COMPLETED,
    DISCARDED,
}
